app/src/main/java/ch/blinkenlights/android/vanilla/PlaylistObserver.java

```diff
@@ -371,7 +384,14 @@ private File dumpAsM3uPlaylist(long id) {
 			if (cursor != null) {
 				pw = new PrintWriter(m3u);
 				while (cursor.moveToNext()) {
-					final String path = cursor.getString(1);
+					// Write paths relative to the playlist export directory, if enabled.
+					final String path;
+					if (mSyncRelativePaths) {
+						path = FileUtils.relativize(mPlaylists,
+							new File(cursor.getString(1)));
+					} else {
+						path = cursor.getString(1);
+					}
```

Comment:
This seems somewhat simpler to read: (At the cost of not using final, but meh..)
```
String path = cursor.getString(1);
 if (mSyncRelativePaths) {
  path = FileUtils.relativize(mPlaylists, new File(path));
 }
```

Reply:
Fixed this issue both here and in FileUtils.resolve.